pkg/rng/entropy.go

```diff
@@ -0,0 +1,139 @@
+package rng
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	trngList map[string]int
+)
+
+const (
+	hwRandomDir                string        = "/sys/class/misc/hw_random"
+	hwRandomCurrentFile        string        = "/sys/class/misc/hw_random/rng_current"
+	hwRandomAvailableFile      string        = "/sys/class/misc/hw_random/rng_available"
+	randomPoolSizeFile         string        = "/proc/sys/kernel/random/poolsize"
+	randomEntropyAvailableFile string        = "/proc/sys/kernel/random/entropy_avail"
+	entropyFeedTime            time.Duration = 10
+	randomDevice               string        = "/dev/random"
+	hwRandomDevice             string        = "/dev/hwrng"
+)
+
+func init() {
```

Comment:
you can remove this function and declare everything at the module's top level like this:

```
var trngList = map[string]int {
    "tpm-rng": 0,
    "intel-rng": 1,
    // and so on
}
```

Reply:
Done